_posts/2017-11-13-tetris-ai-experiment-1-2-single-parent-evolutionary-algorithm.md

```diff
@@ -0,0 +1,258 @@
+
+## Introduction
+
+### Why?
+
+I have been experimenting with artificial intelligence. I was looking for a game. Ideally something with visual output so that I could see the results of my experiments visually. I was inspired to do this by a series of videos I found on youtube about using an evolutionary algorithm to train simple walkers [available here](https://www.youtube.com/watch?v=GOFws_hhZs8&ab_channel=carykh). I have previously looked into AI [here](https://github.com/cohen990/NeuralNetworkTutorial) and [here](https://github.com/cohen990/Coursera); this is the first time I've seen any notable results.
+
+### Theoretical Background
+
+#### Neural Network
+
+Neural networks are used as an analogy for the decision making pathways in a brain.
+
+[![diagram of fully connected neural network]({{site.baseurl}}/assets/img/custom/blog/2017-11-13-tetris-ai-experiment-1-2-single-parent-evolutionary-algorithm/neural-network.png)]({{site.baseurl}}/assets/img/custom/blog/2017-11-13-tetris-ai-experiment-1-2-single-parent-evolutionary-algorithm/neural-network.png)
+
+My network is structured in three layers. The input layer, the hidden layer and the output layer, although other architectures can have an arbitrary number of hidden layers. The advantage of having multiple layers is that higher order features can be studied. For example, if we consider a neural network for analysing images: The first hidden layer can combine the raw data and perhaps understand that a straight line or a corner is a feature in an image. The second hidden layer may be able to combine two straight lines and understand that parallel lines are a feature or combine a straight line and a corner and understand hockey-stick shapes.
+
+In a fully connected neural network, every node in a layer is connected to every node in the following layer. Each node holds a value which is calculated from the previous layer.
+
+The values for the input layer are provided to the network by the game.
+
+The value for each node in the hidden layer is calculated using the following formula:
+
+[![diagram showing the calculations for node activation]({{site.baseurl}}/assets/img/custom/blog/2017-11-13-tetris-ai-experiment-1-2-single-parent-evolutionary-algorithm/activation.gif)]({{site.baseurl}}/assets/img/custom/blog/2017-11-13-tetris-ai-experiment-1-2-single-parent-evolutionary-algorithm/activation.gif)
+
+In English, you take each node in the input layer and multiply them by their weight. You add all of those together and then add a bias. Then you pass that value through the activation function and that is the value of your node.
+
+You calculate the value of each node in the hidden layer in the same way - and then you do the same for the output layer but using the hidden layer as input.
+
+The output layer is read in order to provide the output result of the neural network. Neural networks have different purposes. They are often trying to solve a classification or a regression problem.
+
```

Comment:
While you talk about what a fully connected network is - not sure if you mention what type of connectivity you decided to chose